File: Java/NextPrime.java
import java.util.Scanner;

/**
 * Author: Andrew Jarombek
 * Date: 5/27/2016
 * Next Prime Number - Have the program find prime numbers until the user chooses to stop asking for the next one.
 */
public class NextPrime {

    // Keep track of what the last prime number found/searched for was
    private static int location = 2;

    /**
     * Recursively checks for the next prime number
     * @param num a number to be checked if it is prime
     * @return a prime number
     */
    private static int nextPrime(Integer num) {
        // Update the location to be checked next
        location++;
        if (num == 2) {
            return num;
        } else {
            for (int i = 2; i <= num / 2; i++) {
                if (num % i == 0) {
                    // If this number is not prime, check if number+1 is
                    return nextPrime(location);
                }
            }
            return num;
        }
    }

    /**
     * Program that generates prime numbers in order by the users command
     */
    public static void primeGenerator() {
        System.out.println("Find next Prime? (Y/n): ");
        Scanner scanner = new Scanner(System.in);
        String answer = scanner.nextLine();

        // Check if the user wants another prime (Y) or not (n)
        // Also make sure no illegal commands are entered
        switch (answer) {
            case "Y":
                int nextP = nextPrime(location);
                System.out.println("The Next Prime Is: " + nextP);
                primeGenerator();
                break;
            case "n":
                System.out.println("Goodbye. ");
                break;
            default:
                System.out.println("ERROR: Invalid Command. ");
                primeGenerator();
                break;
        }
    }

    public static void main(String[] args) {
        primeGenerator();
    }
}

File: C/NextPrime.c
#include <stdio.h>
#include <string.h>

// Author: Andrew Jarombek
// Date: 9/15/2016 - 10/8/2016
// Have the program find prime numbers until the user chooses to stop asking for the next one.

#define MAXINPUT 10

int checkInput(char input[], int *prime);
int nextPrime(int *prev);

/* Main method for user input */
int main() {

	int done, prime;

	done = 1;
	prime = 1;

	// While the user hasn't typed in 'n', continue searching for primes
	while (done) {
		char input[MAXINPUT];
		printf("Find the next prime number (Y/n): \n");
		scanf("%s", &input);
		done = checkInput(input, &prime);
	}
}

/* Analyze the user input and choose which action to take */
int checkInput(char input[], int *prime) {
	if (strlen(input) == 1) {
		// If input is 'Y', search for the next prime
		if (input[0] == 'Y') {
			nextPrime(prime);
			return 1;
		// If input is 'N', end search
		} else if (input[0] == 'n') {
			printf("Goodbye.\n");
			return 0;
		// Otherwise give an error message
		} else {
			printf("ERROR: Invalid Command.\n");
			return 1;
		}
	} else {
		printf("ERROR: Invalid Command.\n");
		return 1;
	}
}

/* Find the next prime number */
int nextPrime(int *prev) {
	int found = 0;
	*prev += 1;

	// Base case, 2 is prime
	if (*prev == 2) {
		printf("The next prime is: %d\n", *prev);
	} else {
		// check to see if it is divisible my any numbers besides itself & 1
		for (int i = 2; i <= *prev/2; i++) {
			if (*prev % i == 0) {
				found = nextPrime(prev);
			}
		}

		// This makes sure the prime number is not printed more than once
		// Since there can be many recursive calls occuring 
		if (!found) {
			printf("The next prime is: %d\n", *prev);
			return 1;
		}
	}
}